pkg/executor/importer/import.go

```diff
@@ -1381,6 +1353,27 @@ func (e *LoadDataController) InitDataFiles(ctx context.Context) error {
 		// For non-parquet format, it's always 1.0.
 		sizeExpansionRatio = 1.0
 	)
+
+	checkFirstFile := func(s storage.ExternalStorage, path string) error {
+		var (
+			err      error
+			checkRes *mydump.ParquetPrecheckResult
+		)
+
+		e.detectAndUpdateFormat(path)
+		sourceType := e.getSourceType()
+		if sourceType != mydump.SourceTypeParquet {
+			return nil
+		}
+
+		checkRes, err = mydump.PrecheckParquet(ctx, s, path)
```

Comment:
put those precheck logic into https://github.com/pingcap/tidb/blob/a51c19c1ff5ac9da3cc7adfc811abce34239174e/pkg/executor/importer/precheck.go
and let `disable_precheck` control it